src/Controls/src/Core/Handlers/Items/iOS/ItemsViewController.cs

```diff
@@ -193,18 +194,48 @@ public override void LoadView()
 		public override void ViewWillAppear(bool animated)
 		{
 			base.ViewWillAppear(animated);
-			ConstrainItemsToBounds();
 		}
 
 		public override void ViewWillLayoutSubviews()
 		{
 			ConstrainItemsToBounds();
+
+			if (CollectionView is Items.MauiCollectionView { NeedsCellLayout: true } collectionView)
+			{
+				InvalidateLayoutIfItemsMeasureChanged();
+				collectionView.NeedsCellLayout = false;
+			}
+
 			base.ViewWillLayoutSubviews();
 			InvalidateMeasureIfContentSizeChanged();
 			LayoutEmptyView();
+
+			_laidOut = true;
 		}
 
+		void InvalidateLayoutIfItemsMeasureChanged()
+		{
+			var visibleCells = CollectionView.VisibleCells;
+			List<NSIndexPath> invalidatedPaths = null;
 
+			var visibleCellsLength = visibleCells.Length;
+			for (int n = 0; n < visibleCellsLength; n++)
+			{
+				if (visibleCells[n] is TemplatedCell { MeasureInvalidated: true } cell)
```

Comment:
WHat if its a user UICollectionView custom cell ? why is it important to see if MeasureInvalidated ?